packages/reporting/src/patterns.js

```diff
@@ -299,6 +276,48 @@ export function lookupBuiltinTransform(name) {
   throw new UnsupportedTransformationError(`Unknown transformation: "${name}"`);
 }
 
+/**
+ * Defines the version of the engine that processes the patterns DSL
+ * (Domain Specific Language).
+ *
+ * It is always safe to increase it: neither will it lead to overhead,
+ * nor will it split the population. Its sole purpose is to be used
+ * to disable clients that do not meet the minimum requirements of the
+ * current patterns.
+ */
+const PATTERN_DSL_VERSION = 1;
+
+/**
+ * "Magic" empty rule set, which exists only if patterns were loaded, but
+ * our engine is too old and does not supported them. Therefore, they are
+ * disabled. Although not a typical error, reaching this state should be
+ * avoided; the client will be dead from the perspective of the server.
+ *
+ * There are two to exit this state:
+ * 1) The client updates to a newer version (this is the preferred one)
+ * 2) The server can decide to serve a backward-compatible set of rules to
+ *    restore support for older clients. It will restore the traffic since
+ *    old clients will still poll for patterns.
+ *
+ * Note that option two will not be sustainable over a longer period. It
+ * also comes with the disadvantage that old clients will form their own
+ * group; anonymity will suffer if their population becomes too small.
+ */
+const RULES_REJECTED__ENGINE_TOO_OLD = {};
```

Comment:
maybe a symbol would work better here?

Reply:
This here is use a valid pattern object. The alternative would be to keep the state in a separate variable.